app/firebase/functions/src/posts/posts.processing.ts

```diff
@@ -165,22 +138,27 @@ export class PostsProcessing {
 
     const appPostFull = await this.getPostFull(postId, manager, true);
 
-    // posts without authors does not have mirrors
-    if (!appPostFull.authorId) {
-      return;
-    }
-
-    const user = await this.users.repo.getUser(
-      appPostFull.authorId,
-      manager,
-      true
-    );
-
     /**
-     * Create platformPosts as drafts on all platforms
+     * Create platformPosts as drafts on all platforms (nanopub only for now)
      * */
     const drafts = await Promise.all(
       ([PLATFORM.Nanopub] as PUBLISHABLE_PLATFORM[]).map(async (platformId) => {
+        const authorProfile = await this.users.profiles.getByProfileId(
+          appPostFull.authorProfileId,
+          manager,
+          true
+        );
+
+        if (!authorProfile.userId) {
+          throw new Error('Profile must be of a signed up userId');
```

Comment:
doesn't this need to support non signed up users? since the nanopub platform post needs to get created for each app post?